lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3761,28 +3750,16 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
       builder.createCopyAddr(adjLoc, source, dest, IsTake, IsInitialization);
     }
 
-    // Deallocate adjoint buffers for original indirect parameters.
-    for (auto adjBuf : reversed(origIndirectParamAdjointBuffers)) {
-      if (auto *cleanup = adjBuf.getCleanup())
-        cleanup->applyRecursively(builder, adjLoc);
-      builder.createDeallocStack(adjLoc, adjBuf);
+    // Deallocate local allocations.
+    for (auto *alloc : localAllocations) {
```

Comment:
`localAllocations` are no longer deallocated in reverse order because `localAllocBuilder` creates new adjoint buffers *before* the last allocated adjoint buffer. This is due to a quirk of `SILBuilder`: insertion point can only be set such that new instructions come before an instruction, not after.

This is all fine is `getAdjointBuffer` is the only code appending to `localAllocations`, but there is one other raw call to `localAllocations.push_back(adjBuf);` for the seed adjoint. But actually, I think that raw call should be removed - let me try to remove it now.

Reply:
The only raw call to `localAllocations.push_back` is removed in https://github.com/apple/swift/pull/22645/commits/e90c5b7288ec26a438c59ff9c62ee47d6b9c8cec.

Now, it's fine that `localAllocations` are deallocated in forward order (because there's only one place where `localAllocations` is appended and we can reason about the ordering).